File: pkg/dev_compiler/test/codegen_expected/corelib/date_time4_test.js
dart_library.library('corelib/date_time4_test', null, /* Imports */[
  'dart_sdk',
  'expect'
], function load__date_time4_test(exports, dart_sdk, expect) {
  'use strict';
  const core = dart_sdk.core;
  const dart = dart_sdk.dart;
  const dartx = dart_sdk.dartx;
  const expect$ = expect.expect;
  const date_time4_test = Object.create(null);
  let VoidTodynamic = () => (VoidTodynamic = dart.constFn(dart.definiteFunctionType(dart.dynamic, [])))();
  let VoidTovoid = () => (VoidTovoid = dart.constFn(dart.definiteFunctionType(dart.void, [])))();
  dart.copyProperties(date_time4_test, {
    get supportsMicroseconds() {
      return new core.DateTime.fromMicrosecondsSinceEpoch(1).microsecondsSinceEpoch == 1;
    }
  });
  date_time4_test.main = function() {
    if (dart.test(date_time4_test.supportsMicroseconds)) {
      date_time4_test.testMicrosecondPrecision();
    } else {
      date_time4_test.testMillisecondPrecision();
    }
  };
  dart.fn(date_time4_test.main, VoidTodynamic());
  date_time4_test.testMillisecondPrecision = function() {
    let dt1 = core.DateTime.parse("1999-01-02 23:59:59.999519");
    expect$.Expect.equals(1999, dt1.year);
    expect$.Expect.equals(1, dt1.month);
    expect$.Expect.equals(3, dt1.day);
    expect$.Expect.equals(0, dt1.hour);
    expect$.Expect.equals(0, dt1.minute);
    expect$.Expect.equals(0, dt1.second);
    expect$.Expect.equals(0, dt1.millisecond);
    expect$.Expect.equals(false, dt1.isUtc);
    dt1 = core.DateTime.parse("1999-01-02 23:58:59.999519Z");
    expect$.Expect.equals(1999, dt1.year);
    expect$.Expect.equals(1, dt1.month);
    expect$.Expect.equals(2, dt1.day);
    expect$.Expect.equals(23, dt1.hour);
    expect$.Expect.equals(59, dt1.minute);
    expect$.Expect.equals(0, dt1.second);
    expect$.Expect.equals(0, dt1.millisecond);
    expect$.Expect.equals(true, dt1.isUtc);
    dt1 = core.DateTime.parse("0009-09-09 09:09:09.009411Z");
    expect$.Expect.equals(9, dt1.year);
    expect$.Expect.equals(9, dt1.month);
    expect$.Expect.equals(9, dt1.day);
    expect$.Expect.equals(9, dt1.hour);
    expect$.Expect.equals(9, dt1.minute);
    expect$.Expect.equals(9, dt1.second);
    expect$.Expect.equals(9, dt1.millisecond);
    expect$.Expect.equals(true, dt1.isUtc);
    let svnDate = "2012-03-30T04:28:13.752341Z";
    dt1 = core.DateTime.parse(svnDate);
    expect$.Expect.equals(2012, dt1.year);
    expect$.Expect.equals(3, dt1.month);
    expect$.Expect.equals(30, dt1.day);
    expect$.Expect.equals(4, dt1.hour);
    expect$.Expect.equals(28, dt1.minute);
    expect$.Expect.equals(13, dt1.second);
    expect$.Expect.equals(752, dt1.millisecond);
    expect$.Expect.equals(true, dt1.isUtc);
  };
  dart.fn(date_time4_test.testMillisecondPrecision, VoidTovoid());
  date_time4_test.testMicrosecondPrecision = function() {
    let dt1 = core.DateTime.parse("1999-01-02 23:59:59.999519");
    expect$.Expect.equals(1999, dt1.year);
    expect$.Expect.equals(1, dt1.month);
    expect$.Expect.equals(2, dt1.day);
    expect$.Expect.equals(23, dt1.hour);
    expect$.Expect.equals(59, dt1.minute);
    expect$.Expect.equals(59, dt1.second);
    expect$.Expect.equals(999, dt1.millisecond);
    expect$.Expect.equals(519, dt1.microsecond);
    expect$.Expect.equals(false, dt1.isUtc);
    dt1 = core.DateTime.parse("1999-01-02 23:58:59.999519Z");
    expect$.Expect.equals(1999, dt1.year);
    expect$.Expect.equals(1, dt1.month);
    expect$.Expect.equals(2, dt1.day);
    expect$.Expect.equals(23, dt1.hour);
    expect$.Expect.equals(58, dt1.minute);
    expect$.Expect.equals(59, dt1.second);
    expect$.Expect.equals(999, dt1.millisecond);
    expect$.Expect.equals(519, dt1.microsecond);
    expect$.Expect.equals(true, dt1.isUtc);
    dt1 = core.DateTime.parse("0009-09-09 09:09:09.009411Z");
    expect$.Expect.equals(9, dt1.year);
    expect$.Expect.equals(9, dt1.month);
    expect$.Expect.equals(9, dt1.day);
    expect$.Expect.equals(9, dt1.hour);
    expect$.Expect.equals(9, dt1.minute);
    expect$.Expect.equals(9, dt1.second);
    expect$.Expect.equals(9, dt1.millisecond);
    expect$.Expect.equals(411, dt1.microsecond);
    expect$.Expect.equals(true, dt1.isUtc);
    let svnDate = "2012-03-30T04:28:13.752341Z";
    dt1 = core.DateTime.parse(svnDate);
    expect$.Expect.equals(2012, dt1.year);
    expect$.Expect.equals(3, dt1.month);
    expect$.Expect.equals(30, dt1.day);
    expect$.Expect.equals(4, dt1.hour);
    expect$.Expect.equals(28, dt1.minute);
    expect$.Expect.equals(13, dt1.second);
    expect$.Expect.equals(752, dt1.millisecond);
    expect$.Expect.equals(341, dt1.microsecond);
    expect$.Expect.equals(true, dt1.isUtc);
  };
  dart.fn(date_time4_test.testMicrosecondPrecision, VoidTovoid());
  // Exports:
  exports.date_time4_test = date_time4_test;
});

File: tests/corelib_2/date_time4_test.dart
// Copyright (c) 2012, the Dart project authors.  Please see the AUTHORS file
// for details. All rights reserved. Use of this source code is governed by a
// BSD-style license that can be found in the LICENSE file.

import "package:expect/expect.dart";

// Test fromString with 6 digits after the decimal point.

bool get supportsMicroseconds =>
    new DateTime.fromMicrosecondsSinceEpoch(1).microsecondsSinceEpoch == 1;

main() {
  if (supportsMicroseconds) {
    testMicrosecondPrecision();
  } else {
    testMillisecondPrecision();
  }
}

void testMillisecondPrecision() {
  // We only support milliseconds. If the user supplies more data (the "51"
  // here), we round.
  DateTime dt1 = DateTime.parse("1999-01-02 23:59:59.999519");
  Expect.equals(1999, dt1.year);
  Expect.equals(1, dt1.month);
  Expect.equals(3, dt1.day);
  Expect.equals(0, dt1.hour);
  Expect.equals(0, dt1.minute);
  Expect.equals(0, dt1.second);
  Expect.equals(0, dt1.millisecond);
  Expect.equals(false, dt1.isUtc);
  dt1 = DateTime.parse("1999-01-02 23:58:59.999519Z");
  Expect.equals(1999, dt1.year);
  Expect.equals(1, dt1.month);
  Expect.equals(2, dt1.day);
  Expect.equals(23, dt1.hour);
  Expect.equals(59, dt1.minute);
  Expect.equals(0, dt1.second);
  Expect.equals(0, dt1.millisecond);
  Expect.equals(true, dt1.isUtc);
  dt1 = DateTime.parse("0009-09-09 09:09:09.009411Z");
  Expect.equals(9, dt1.year);
  Expect.equals(9, dt1.month);
  Expect.equals(9, dt1.day);
  Expect.equals(9, dt1.hour);
  Expect.equals(9, dt1.minute);
  Expect.equals(9, dt1.second);
  Expect.equals(9, dt1.millisecond);
  Expect.equals(true, dt1.isUtc);
  String svnDate = "2012-03-30T04:28:13.752341Z";
  dt1 = DateTime.parse(svnDate);
  Expect.equals(2012, dt1.year);
  Expect.equals(3, dt1.month);
  Expect.equals(30, dt1.day);
  Expect.equals(4, dt1.hour);
  Expect.equals(28, dt1.minute);
  Expect.equals(13, dt1.second);
  Expect.equals(752, dt1.millisecond);
  Expect.equals(true, dt1.isUtc);
}

void testMicrosecondPrecision() {
  DateTime dt1 = DateTime.parse("1999-01-02 23:59:59.999519");
  Expect.equals(1999, dt1.year);
  Expect.equals(1, dt1.month);
  Expect.equals(2, dt1.day);
  Expect.equals(23, dt1.hour);
  Expect.equals(59, dt1.minute);
  Expect.equals(59, dt1.second);
  Expect.equals(999, dt1.millisecond);
  Expect.equals(519, dt1.microsecond);
  Expect.equals(false, dt1.isUtc);
  dt1 = DateTime.parse("1999-01-02 23:58:59.999519Z");
  Expect.equals(1999, dt1.year);
  Expect.equals(1, dt1.month);
  Expect.equals(2, dt1.day);
  Expect.equals(23, dt1.hour);
  Expect.equals(58, dt1.minute);
  Expect.equals(59, dt1.second);
  Expect.equals(999, dt1.millisecond);
  Expect.equals(519, dt1.microsecond);
  Expect.equals(true, dt1.isUtc);
  dt1 = DateTime.parse("0009-09-09 09:09:09.009411Z");
  Expect.equals(9, dt1.year);
  Expect.equals(9, dt1.month);
  Expect.equals(9, dt1.day);
  Expect.equals(9, dt1.hour);
  Expect.equals(9, dt1.minute);
  Expect.equals(9, dt1.second);
  Expect.equals(9, dt1.millisecond);
  Expect.equals(411, dt1.microsecond);
  Expect.equals(true, dt1.isUtc);
  String svnDate = "2012-03-30T04:28:13.752341Z";
  dt1 = DateTime.parse(svnDate);
  Expect.equals(2012, dt1.year);
  Expect.equals(3, dt1.month);
  Expect.equals(30, dt1.day);
  Expect.equals(4, dt1.hour);
  Expect.equals(28, dt1.minute);
  Expect.equals(13, dt1.second);
  Expect.equals(752, dt1.millisecond);
  Expect.equals(341, dt1.microsecond);
  Expect.equals(true, dt1.isUtc);
}
